src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1996,37 +2093,54 @@ public FormValidation doCheckCredentialsId(@CheckForNull @AncestorInPath Item co
         public FormValidation doValidateRepositoryUrlAndCredentials(@CheckForNull @AncestorInPath Item context,
                                                                     @QueryParameter String repositoryUrl,
                                                                     @QueryParameter String credentialsId) {
+
+            // Do some context permissions checks
             if (context == null && !Jenkins.get().hasPermission(Jenkins.ADMINISTER) ||
                 context != null && !context.hasPermission(Item.EXTENDED_READ)) {
                 return FormValidation.error("Unable to validate repository information"); // not supposed to be seeing this form
             }
             if (context != null && !context.hasPermission(CredentialsProvider.USE_ITEM)) {
                 return FormValidation.error("Unable to validate repository information"); // not permitted to try connecting with these credentials
             }
+
+            // Verify we have a well-formed GitHub API URL
             GitHubRepositoryInfo info;
 
             try {
                 info = GitHubRepositoryInfo.forRepositoryUrl(repositoryUrl);
-            } catch (IllegalArgumentException e) {
+            } catch (Throwable e) {
+                if (e instanceof IllegalArgumentException && e.getCause() != null) {
+                    e = e.getCause();
+                }
```

Comment:
What's this for? If it's to try to make it easier for users to understand the stack trace, I would delete these lines, and change the next line from `e.getMessage` to something like `The format of the repository URL is invalid. Please check the help text and adjust the URL`, which is probably more helpful unless we are dealing with a bug.

Also though, if the changes in `doValidateRepositoryUrlAndCredentials` aren't required for the main fix, it might be better to move it to a separate PR.

Reply:
Okay, you've both said this less than ideal. I see a way to clean it up.